### PR TITLE
Document that the StyleHandler is final now

### PR DESCRIPTION
**Pull request by TimWolla**

### CODE REVIEW

File: docs/migration/wsc53/php.md [hunk 1]
```diff
@@ -80,3 +80,11 @@ If you implemented a custom search engine and relied on this method, you can inl
 You should take the time to verify the rewritten queries against the manual of the search engine to make sure it cannot generate malformed queries or security issues.
 
 See [WoltLab/WCF#3815](https://github.com/WoltLab/WCF/issues/3815) for details.
+
+## Styles
+
+The `StyleCompiler` is marked `final` now.
```
Comment (Gravatronics) [inline docs/migration/wsc53/php.md:86]:
```suggestion
The `StyleCompiler` class is marked `final` now.
```

File: docs/migration/wsc53/php.md [hunk 2]
```diff
@@ -80,3 +80,11 @@ If you implemented a custom search engine and relied on this method, you can inl
...
+It was noticed that the `$compiler` property was a design issue that leaked compiler state across multiple compiled styles, possibly causing misgenerated stylesheets.
```
Comment (Gravatronics) [inline docs/migration/wsc53/php.md:87]:
```suggestion
The internal SCSS compiler object being stored in the `$compiler` property was a design issue that leaked compiler state across multiple compiled styles, possibly causing misgenerated stylesheets.
```

File: docs/migration/wsc53/php.md [hunk 3]
```diff
@@ -80,3 +80,11 @@ If you implemented a custom search engine and relied on this method, you can inl
...
+As the removal of the `$compiler` property effectively broke compatibility within the `StyleCompiler` and as the `StyleCompiler` never was meant to be extended it was marked final.
```
Comment (Gravatronics) [inline docs/migration/wsc53/php.md:88]:
```suggestion
As the removal of the `$compiler` property effectively broke compatibility within the `StyleCompiler` and as the `StyleCompiler` never was meant to be extended, it was marked final.
```

File: docs/migration/wsc53/php.md [hunk 4]
```diff
@@ -80,3 +80,11 @@ If you implemented a custom search engine and relied on this method, you can inl
...
+
+See [WoltLab/WCF#3929](https://github.com/WoltLab/WCF/pull/3929) for details.
```
Comment (Gravatronics) [inline docs/migration/wsc53/php.md:90]:
```suggestion
Refer to [WoltLab/WCF#3929](https://github.com/WoltLab/WCF/pull/3929) for details.
```

**Review by Gravatronics — CHANGES_REQUESTED** [review]

File: docs/migration/wsc53/php.md (hunk 4 above)
Comment (TimWolla) [reply docs/migration/wsc53/php.md:90]: This one should be consistently adjusted everywhere.

**Pull request merged**